packages/boot/test/fast-usdc/fast-usdc.test.ts

```diff
@@ -575,11 +492,103 @@ test.serial('makes usdc advance', async t => {
   await documentStorageSchema(t, storage, doc);
 });
 
-test.serial('writes pool metrics to vstorage', async t => {
-  const { storage } = t.context;
+test.serial('minted before observed; forward path', async t => {
+  const { bridgeUtils, storage } = t.context;
+
+  // Mock user data
+  const EUD = 'agoric1usermintedbeforeobserved';
+  const lastNodeValue = storage.getValues('published.fastUsdc').at(-1);
+  const { settlementAccount, nobleICA } = JSON.parse(NonNullish(lastNodeValue));
+
+  // Create mock evidence
+  const mintAmt = 500_000n; // 0.5 USDC
+  const recipientAddress = encodeAddressHook(settlementAccount, { EUD });
+  const evidence = MockCctpTxEvidences.AGORIC_PLUS_AGORIC(recipientAddress);
+  evidence.tx.amount = mintAmt;
+  t.log('USDC minted before evidence is observed');
+
+  // Simulate USDC being minted and received BEFORE evidence is submitted
+  await bridgeUtils.runInbound(
+    BridgeId.VTRANSFER,
+    buildVTransferEvent({
+      sequence: '1',
+      amount: evidence.tx.amount,
+      denom: 'uusdc',
+      sender: evidence.tx.forwardingAddress,
+      target: settlementAccount,
+      receiver: recipientAddress,
+      sourceChannel: nobleToAgoricChannel,
+      destinationChannel: agoricToNobleChannel,
+    }),
+  );
+
+  await eventLoopIteration();
+  t.log('USDC funds received in settlement account before evidence submission');
+
+  // Now submit evidence from oracles (after funds already received)
+  const { walletFactoryDriver: wfd } = t.context;
+  const oracles = await Promise.all(
+    oracleAddrs.map(addr => wfd.provideSmartWallet(addr)),
+  );
+  await Promise.all(
+    oracles.map(wallet =>
+      wallet.sendOffer({
+        id: 'submit-evidence-after-mint',
+        invitationSpec: {
+          source: 'continuing',
+          previousOffer: 'claim-oracle-invitation',
+          invitationMakerName: 'SubmitEvidence',
+          invitationArgs: [evidence],
+        },
+        proposal: {},
+      }),
+    ),
+  );
+
+  await eventLoopIteration();
+
+  await bridgeUtils.runInbound(
+    BridgeId.VTRANSFER,
+    buildVTransferEvent({
+      sender: settlementAccount,
+      target: settlementAccount,
+      receiver: nobleICA,
+      sourceChannel: agoricToNobleChannel,
+      destinationChannel: nobleToAgoricChannel,
+      sequence: '2',
+      amount: evidence.tx.amount,
+      memo: JSON.stringify(
+        /** @type {ForwardInfo} */
+        {
+          forward: {
+            receiver: recipientAddress,
+            port: 'transfer',
+            channel:
+              fetchedChainInfo.noble.connections['agoric-3'].transferChannel
+                .channelId,
+          },
+        },
+      ),
+    }),
+  );
+
+  // Verify the transaction status was set to FORWARDED
+  const getTxStatus = txHash =>
```

Comment:
nit but this comment doesn't have to do with the function. consider moving the function up top as a scoped helper